x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/EngineIndexService.java

```diff
@@ -308,7 +303,47 @@ public void deleteEngine(String engineName, ActionListener<DeleteResponse> liste
         } catch (Exception e) {
             listener.onFailure(e);
         }
+    }
+
+    GetAliasesResponse getAlias(String engineAliasName) {
+        return clientWithOrigin.admin().indices().getAliases(new GetAliasesRequest(engineAliasName)).actionGet();
+    }
+
+    private void removeAlias(String engineAliasName, ActionListener<AcknowledgedResponse> listener) {
+        IndicesAliasesRequest aliasesRequest = new IndicesAliasesRequest().addAliasAction(
+            IndicesAliasesRequest.AliasActions.remove().aliases(engineAliasName).indices("*")
+        );
+        clientWithOrigin.admin().indices().aliases(aliasesRequest, ActionListener.wrap(response -> {
+            if (response.isAcknowledged() == false) {
+                logger.warn("Request to remove alias [{}] response was not acknowledged", engineAliasName);
+            }
+            listener.onResponse(null);
```

Comment:
Why not providing the `response` to the listener? It sounds weird to pass `null` as parameter to `onResponse`

Reply:
Good suggestion, changed!